waiter/src/waiter/util/utils.clj

```diff
@@ -362,18 +389,18 @@
   (cond-> response
     error-class (assoc :error-class error-class)))
 
-(defn data->error-response
-  "Converts the provided data map into a ring response.
+(defn- data-render-fn->error-response
+  "Converts the provided data map and render functions into a ring response.
    The data map is expected to contain the following keys: details, headers, message, and status."
-  [{:keys [details headers status] :or {status http-400-bad-request} :as data-map} request]
-  (let [error-context (build-error-context data-map request)
+  [{:keys [details headers status] :or {status http-400-bad-request} :as data-map} request build-context-fn render-html-fn render-text-fn]
```

Comment:
Nitpick: I'd prefer to keep `request` as the last argument and move the render functions as the first couple of arguments.